### PR TITLE
Issue 333 pre commit hooks

### PR DESCRIPTION
**Pull request by tdonaworth**
## What changed

* Added `.pre-commit-config.yaml` along with a few early default pre-commit actions
* Updated the Readme to include pre-commit instructions
* Made a few remediations as a result of newly added pre-commit actions

## Issue

#333 

## How to test

All pre-commit actions can be run locally with `pre-commit run --all-files`

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  - repo: https://github.com/hadolint/hadolint
+    rev: v2.10.0
+    hooks:
+      - id: hadolint
+
+  - repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+    -   id: black
```
Comment (halprin) [inline .pre-commit-config.yaml:30]: Is there a way to run a local command line?  We already have a way to do this linting (and more) locally.  `pipenv run nox -s black`, `pipenv run nox -s lint`, and ` yarn lint`.

I rather run the same thing we would run locally.

File: README.md [hunk 2]
```diff
@@ -150,4 +150,24 @@ docker run -it --rm -v "$(pwd)/docs":/work -w /work \
...
+* [check-json](https://github.com/pre-commit/pre-commit-hooks#check-json)
+* [check-yaml](https://github.com/pre-commit/pre-commit-hooks#check-yaml)
+* [check-added-large-files](https://github.com/pre-commit/pre-commit-hooks#check-added-large-files)
+* [check-merge-conflict](https://github.com/pre-commit/pre-commit-hooks#check-merge-conflict)
+
+### IBM's [Detect-Secrets](https://github.com/ibm/detect-secrets)
+
+### [Hadolint](https://github.com/hadolint/hadolint) Dockerfile linting
+
+### Python [Black](https://github.com/psf/black)
```
Comment (halprin) [inline README.md:173]: Do we have nothing to say under these sections?

File: docker-compose.yml [hunk 3]
```diff
@@ -6,17 +6,24 @@ services:
     image: "postgres:14.4"
     environment:
       - POSTGRES_PASSWORD=local_password
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      interval: 5s
+      timeout: 5s
+      retries: 5
```
Comment (halprin) [inline docker-compose.yml:13]: Love it!

**Review by halprin — COMMENTED** [review]
Nice!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (tdonaworth) [reply .pre-commit-config.yaml:30]: I guess it simply depends on personal usability, this simply forces black to run, if we're not doing any extra black-specific config, then we can go either way.

File: README.md (hunk 2 above)
Comment (tdonaworth) [reply README.md]: added some additional context

File: .pre-commit-config.yaml (hunk 1 above)
Comment (tdonaworth) [reply .pre-commit-config.yaml]: updated pre-commit to use `pipenv run nox -s black`

File: Pipfile [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+url = "https://pypi.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+
+[dev-packages]
+
+[requires]
+python_version = "3.9"
```
Comment (halprin) [inline Pipfile:11]: I don't think you meant to commit this file.

**Review by halprin — APPROVED** [review]

**Pull request merged**